ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -179,6 +190,30 @@ private void handleTouchEvent(MotionEvent ev) {
     }
   }
 
+  /**
+   * Calculate view offset relative to this root view. This offset can be used to convert raw event
+   * coordinates to local coordinates with respect to the target view.
+   */
+  private void getOffsetOfView(View view, int[] outputOffset) {
+    int[] rootViewLocationInScreen = {0, 0};
+    getLocationOnScreen(rootViewLocationInScreen);
+    view.getLocationOnScreen(outputOffset);
+    outputOffset[0] -= rootViewLocationInScreen[0];
+    outputOffset[1] -= rootViewLocationInScreen[1];
+  }
+
+  /**
+   * Convert an event to local coordinates relative to the target view. A modified copy of the
+   * original event is returned.
+   */
+  private MotionEvent convertEventToLocal(MotionEvent event, View targetView) {
+    MotionEvent evCopy = MotionEvent.obtain(event);
+    int[] viewOffset = {0, 0};
+    getOffsetOfView(targetView, viewOffset);
```

Comment:
Why return the output in the second argument if you do an allocation anyway? Could just return it from the method. But then we really should to avoid the allocation.